library/objective-c/EnvoyEngineImpl.m

```diff
@@ -29,13 +119,33 @@ - (void)dealloc {
   [[NSNotificationCenter defaultCenter] removeObserver:self];
 }
 
+- (int)registerFilter:(EnvoyHTTPFilter *)filter {
+  // TODO(goaway): Everything here leaks, but it's all be tied to the life of the engine.
+  // This will need to be updated for https://github.com/lyft/envoy-mobile/issues/332
+  ios_http_filter_context *context = safe_malloc(sizeof(ios_http_filter_context));
+  CFBridgingRetain(filter);
+  context->filter = filter;
+  envoy_http_filter *api = safe_malloc(sizeof(envoy_http_filter));
+  api->on_request_headers = ios_http_filter_on_request_headers;
+  api->on_request_data = NULL;
+  api->on_response_headers = ios_http_filter_on_response_headers;
+  api->on_response_data = NULL;
+  api->context = context;
+  register_platform_api(filter.name.UTF8String, api);
+  return 0;
```

Comment:
`ENVOY_SUCCESS`?

Reply:
We don't have the type available on the header here, so we've only used `0` and `1` in this file so far. The extern-based solution I propose in this PR for filter status would work for this as well though, so maybe I can fix this for all the outstanding cases in a subsequent PR?

Reply:
SGTM